### PR TITLE
docs: add Typescript Standards

### PR DESCRIPTION
**Pull request by timouano**
## What does this PR do?
- Introduces Typescript standards

## Why did you choose this solution?
- To ensure code quality and consistency

## What is the dependency graph of the changes?
- No dependencies

## How to test/verify?
- Review the Typescript standards document

## Related ticket url:
-

### CODE REVIEW

**Comment by symphdevtools** [comment]
Hi! I'm generating PR title and description for you and will automatically apply it to your PR. I'll react :rocket: to this comment when done. Just sit back and chill!

When you have new changes to this PR and you want to regenerate the title and PR description, just comment `churva`!

Let me know if something is not working well, e.g.:
- Some parts of the title and description weren't preserved when generating title and description
- Disappearing screenshots or links
- Etc.

Thanks!

(RIP hatdog)

File: standards/typescript_standards.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+};
+```
+
+Some properties can only be modified when an object is first created. You can specify this by putting `readonly` before the name of the property, as the following example shows.
+```
+interface Employee {
+  readonly idNumber: number;
+  readonly age: number;
+}
+```
```
Comment (JHNLWHD) [inline standards/typescript_standards.md]: @timouano let's confirm this, afaik we prefer `types` cc: @johnreyev @coderaven

**Review by JHNLWHD — CHANGES_REQUESTED** [review]
@timouano let's also add the `strict: true` rule in tsconfig to ensure and remind devs that their projects are using typescript

File: standards/typescript_standards.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+let data: string;
+```
+
+## Leverage Types for Object Shapes
+Types in TypeScript are powerful tools for defining the shapes of objects. Use Types to ensure objects have the expected structure. In the example below, a Type is used to standardize the users to ensure that constants provide the expected structure.
+```
+Type User = {
+  fullName: string;
+  age: number;
+}
```
Comment (JHNLWHD) [inline standards/typescript_standards.md]: This is invalid syntax @timouano - it should be `type` - update all references

**Review by JHNLWHD — APPROVED** [review]
LGTM 👍🏼

**Pull request merged**